src/components/ChainSelector.tsx

```diff
@@ -0,0 +1,198 @@
+import { SimpleDropdown, SimpleListDropdownMenu } from '~/components/SimpleDropdown'
+import UnstyledNetworkIcon from '~/shared/components/NetworkIcon'
+import SvgIcon from '~/shared/components/SvgIcon'
+import { useAvailableChains, useChainStore, useCurrentChain } from '~/shared/stores/chain'
+import { Chain } from '~/types'
+import { COLORS, LAPTOP } from '~/shared/utils/styled'
+
+type MenuItemProps = {
+    chain: Chain
+    isSelected: boolean
+    onClick: () => void
+}
+
+const MenuItem = ({ chain, isSelected, onClick }: MenuItemProps) => (
+    <MenuItemContainer onClick={onClick}>
+        <NetworkIcon chainId={chain.id} />
+        <div>{chain.name}</div>
+        {isSelected ? <SvgIcon name="tick" /> : <div />}
+    </MenuItemContainer>
+)
+
+type MenuProps = {
+    chains: Chain[]
+    selectedChain: Chain
+    toggle: (value: boolean) => void
+}
+
+const Menu = ({ chains, selectedChain, toggle }: MenuProps) => {
+    const setSelectedChain = useChainStore((state) => state.setSelectedChain)
+
+    return (
+        <MenuContainer>
+            <ChainGrid>
+                {chains.map((c) => (
+                    <MenuItem
+                        key={c.id}
+                        chain={c}
+                        isSelected={c.id === selectedChain.id}
+                        onClick={() => {
+                            setSelectedChain(c.id)
+                            toggle(false)
+                        }}
+                    />
+                ))}
+            </ChainGrid>
+        </MenuContainer>
+    )
+}
+
+interface Props {
+    menuAlignment: 'left' | 'right'
+}
+
+export const ChainSelector = ({ menuAlignment = 'left', ...props }: Props) => {
+    const availableChains = useAvailableChains()
+    const selectedChain = useCurrentChain()
+
+    return (
+        <>
+            <SimpleDropdown
+                menu={(toggle) => (
+                    <Menu
+                        chains={availableChains}
+                        selectedChain={selectedChain}
+                        toggle={toggle}
+                    />
+                )}
+                align={menuAlignment}
+                {...props}
+            >
+                {(toggle, isOpen) => (
+                    <Toggle $isOpen={isOpen} onClick={() => toggle((v) => !v)}>
+                        <NetworkIcon chainId={selectedChain.id} />
+                        <div>{selectedChain.name}</div>
+                        <Caret name="caretUp" $isOpen={isOpen} />
+                    </Toggle>
+                )}
+            </SimpleDropdown>
+        </>
+    )
+}
+
+const MenuContainer = styled(SimpleListDropdownMenu)`
+    color: ${COLORS.primaryLight};
+    max-width: 100%;
+`
+
+const ChainGrid = styled.div`
+    display: grid;
+    grid-template-columns: auto auto auto;
+    grid-auto-rows: 52px;
+    gap: 0px 8px;
+`
+
+const MenuItemContainer = styled.div`
+    display: grid;
+    grid-template-columns: subgrid;
```

Comment:
BTW, I used the new and shiny `subgrid` feature here. It will share the grid tracks from parent. This way the tick mark will be shown correctly and aligned by longest chain name.

We could use this feature also in our `<ScrollTable>s`. I think there were some hacks to get the columns aligned properly.